torchrl/envs/libs/vmas.py

```diff
@@ -0,0 +1,428 @@
+from typing import Dict, List, Optional
+
+import torch
+from tensordict.tensordict import TensorDict, TensorDictBase
+from torchrl.data import CompositeSpec, UnboundedContinuousTensorSpec
+from torchrl.envs.common import _EnvWrapper
+from torchrl.envs.libs.gym import _gym_to_torchrl_spec_transform
+
+try:
+    import vmas
+
+    _has_vmas = True
+
+except ImportError as err:
+
+    _has_vmas = False
+    IMPORT_ERR = str(err)
+
+__all__ = ["VmasWrapper", "VmasEnv"]
+
+
+def _get_envs() -> List:
+    if not _has_vmas:
+        return []
+    all_scenarios = vmas.scenarios + vmas.mpe_scenarios + vmas.debug_scenarios
+    # TODO heterogenous spaces
+    # For now torchrl does not support heterogenous spaces (Tple(Box)) so many OpenAI MPE scenarios do not work
+    heterogenous_spaces_scenarios = [
+        "simple_adversary",
+        "simple_crypto",
+        "simple_push",
+        "simple_speaker_listener",
+        "simple_tag",
+        "simple_world_comm",
+    ]
+
+    return [
+        scenario
+        for scenario in all_scenarios
+        if scenario not in heterogenous_spaces_scenarios
+    ]
+
+
+def _selective_unsqueeze(tensor: torch.Tensor, batch_size: torch.Size, dim: int = -1):
```

Comment:
I think this would be useful for other torchrl components (done, reward and sometimes log-probs need to be unsqueezed).
Can we move this to envs.utils or such?

Reply:
yes will do